scripts/test-infra/test-e2e-presubmit.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash -e
+set -o pipefail
+
+# Configure environment
+export IMAGE_REGISTRY="localhost:5001"
+export CLUSTER_NAME=$(git describe --tags --dirty --always)
+export KUBECONFIG="/tmp/kubeconfig_$CLUSTER_NAME"
+
+# create registry container unless it already exists
+reg_name='kind-registry'
+reg_port='5001'
+if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
+    registry:2
+fi
+
+# create a cluster with the local registry enabled in containerd
+cat <<EOF | kind create cluster --kubeconfig $KUBECONFIG --name $CLUSTER_NAME --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
+    endpoint = ["http://${reg_name}:5000"]
+EOF
+
+# connect the registry to the cluster network if not already connected
+if [ "$(docker inspect -f='{{json .NetworkSettings.Networks.kind}}' "${reg_name}")" = 'null' ]; then
+  docker network connect "kind" "${reg_name}"
+fi
+
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-registry-hosting
+  namespace: kube-public
+data:
+  localRegistryHosting.v1: |
+    host: "localhost:${reg_port}"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+EOF
+
+# build the local image
+make image
+
+# push the image to the local registry
+make push
+
+# run the tests
+make e2e-test
+
+# clean up the environment
+kind delete cluster --kubeconfig $KUBECONFIG --name $CLUSTER_NAME
```

Comment:
Do we want to stop/delete the docker registry too?

Reply:
yep, added it